server/main.py

```diff
@@ -201,14 +212,21 @@ def update_code_cache(self, room_code: str, replace_data: ReplaceData) -> None:
                 updated_code = current_code[:from_index] + new_value + current_code[to_index:]
                 self._rooms[room_code]["code"] = updated_code
 
-    async def broadcast(self, data: EventResponse, room_code: str, sender: Client | None = None) -> None:
+    async def broadcast(
+        self, data: SyncData | ReplaceData, room_code: str, sender: Client | None = None, buggy: bool = False
+    ) -> None:
         """Broadcasts data to all active connections.
 
         Args:
             data: The data to be sent to the clients.
             room_code: The room to which the data will be sent.
             sender (optional): The client who sent the message.
+            buggy (optional): To send back modified code.
         """
+        if buggy:
+            data = self._modify_code(room_code)
+            self.update_code_cache(room_code, data)
+
```

Comment:
As I said in a previous comment, I probably wouldn't leave all this "work" to `broadcast`. For me, `broadcast` should just send the data to other clients and nothing else.

That said, I was thinking about moving all the editor-related logic (e.g. modifying code) to another class.

Reply:
Wouldn't that just be the Modifier class? Could you give a small example of how you would approach it?

Reply:
That is currently the `ConnectionManager` class: it stores the code and can update and modify it (for the modifications it depends on the `Modifier` class obviously).

My idea would be having something like an `Editor` class that does all of this:
```py
class Editor:
    def __init__(...):
        self.code = ...
        ...
    def update_code_cache(...):
        ...
    def modify_code(...):
        ...
```

Reply:
Ok, that already looks like it makes more logical sense, especially with the naming of functions.
`ConnectionManager.update_code_cache()` vs `Editor.update_code_cache()`. Shall I add it to this PR or a new one?

Reply:
> Ok, that already looks like it makes more logical sense, especially with the naming of functions. `ConnectionManager.update_code_cache()` vs `Editor.update_code_cache()`. Shall I add it to this PR or a new one?

Maybe a new one, it could get bigger I think.